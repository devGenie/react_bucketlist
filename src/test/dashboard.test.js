import React from 'react';
import { mount, shallow } from 'enzyme';
import { stub, spy } from 'sinon';
import toJson from 'enzyme-to-json';
import 'should';
import 'should-enzyme';
import fetchMock from 'fetch-mock';
import localStorageMock from '../__mocks__/localStorage';
import DashBoard from '../components/dashboard';
import bucketlistmock from '../__mocks__/bucketlists-mock';
import Bucketlists from '../components/bucketlists';

window.sessionStorage = localStorageMock;

describe('<DashBoard>', () => {
  it('calls componentDidMount', () => {
      spy(DashBoard.prototype, 'componentDidMount');
      const wrapper=mount(<DashBoard />);
      expect(DashBoard.prototype.componentDidMount.calledOnce).toBe(true);
    });

    it('renders bucketlists',()=>{
      const wrapper=mount(<Bucketlists data={bucketlistmock['data']} />)
      expect(wrapper.find('.bucketlist')).toHaveLength(4);
    })

    it('renders correctly', ()=>{
        const wrapper=shallow(<DashBoard/>);
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});