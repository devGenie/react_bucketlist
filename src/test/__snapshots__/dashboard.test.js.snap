// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashBoard> renders correctly 1`] = `
<div>
  <Topnav
    searchResults={[Function]}
  />
  <div
    className="container"
  >
    <div
      className="row"
      id="bklist_controller"
    >
      <AddButton />
    </div>
    <Spinners
      status="show"
    />
    <BucketLists
      data={Array []}
      formCallback={[Function]}
      handleDelete={[Function]}
      itemEditCallback={[Function]}
      load={[Function]}
      pages={0}
    />
  </div>
  <BucketlistForm
    handler={[Function]}
  />
  <ItemForm
    caller={[Function]}
    onComplete={[Function]}
  />
  <EditBucketlist
    caller={[Function]}
    editing={
      Object {
        "description": "",
        "name": "",
      }
    }
    onComplete={[Function]}
  />
  <EditItem
    caller={
      Object {
        "bucketlist": "",
        "callback": "",
        "item": Object {
          "id": "",
          "name": "",
        },
      }
    }
  />
</div>
`;
